asdf/generic_io.py

```diff
@@ -101,8 +101,7 @@ def _array_tofile_chunked(write, array, chunksize):  # pragma: no cover
 def _array_tofile_simple(fd, write, array):
     return write(array.data)
 
-
-if sys.platform == 'darwin':  # pragma: no cover
+if sys.platform == 'darwin':
```

Comment:
Odd to have this logic at the module level.  Wouldn't it be clearer to have the logic in the calling function?  Or are there more than one calling functions?

Reply:
I don't know for sure, but I suspect whoever wrote this was trying to be efficient by doing the check once at import time instead of on every call.